TidepoolOnboarding/View Models/OnboardingViewModel.swift

```diff
@@ -103,6 +121,26 @@ class OnboardingViewModel: ObservableObject, CGMManagerCreateNotifying, CGMManag
         }
     }
     
+    func updateLastAccessedDate() {
+        if shouldRestart {
+            restart()
+        }
+
+        self.lastAccessDate = Date()
+    }
+
+    private let lastAccessDurationMaximum: TimeInterval = .days(7)
+
+    private var shouldRestart: Bool { canRestart && wantRestart }
+
+    private var canRestart: Bool { prescription == nil }
+
+    private var wantRestart: Bool { lastAccessDate.addingTimeInterval(lastAccessDurationMaximum) < Date() }
```

Comment:
[nit] this variable name is a bit odd for what it represents.  Should it be `hasExpired`?

Reply:
It represents whether onboarding wants to restart because the 7-day limit is passed. So, `wantRestart` seems right to me.

Reply:
Hm, again, sorry if I'm nitpicking, but again I'm not familiar with this limit, so how about `timeLimitExpired`?